scanner/src/main/kotlin/Scanner.kt

```diff
@@ -203,16 +203,30 @@ class Scanner(
             ).filterByVcsPath()
         }
 
-        val relevantProvenances = provenances.flatMapTo(mutableSetOf()) {
-            it.getKnownProvenancesWithoutVcsPath().values
+        val vcsPathsForProvenances = buildMap<KnownProvenance, MutableSet<String>> {
+            provenances.forEach { provenance ->
+                val packageVcsPath = provenance.packageProvenance?.vcsPath.orEmpty()
+
+                provenance.getKnownProvenancesWithoutVcsPath().forEach { (repositoryPath, provenance) ->
+                    getVcsPathForRepositoryOrNull(packageVcsPath, repositoryPath)?.let { vcsPath ->
+                        getOrPut(provenance) { mutableSetOf() } += vcsPath
+                    }
+                }
+            }
         }
 
         val scanResults = controller.getAllScanResults().map { scanResult ->
             scanResult.copy(
                 provenance = scanResult.provenance.alignRevisions(),
                 summary = scanResult.summary.copy(packageVerificationCode = "")
             )
-        }.filterTo(mutableSetOf()) { scanResult -> scanResult.provenance in relevantProvenances }
+        }.filterTo(mutableSetOf()) { scanResult ->
+            scanResult.provenance in vcsPathsForProvenances.keys
+        }.mapTo(mutableSetOf()) { scanResult ->
+            scanResult.copy(
+                summary = scanResult.summary.filterByPaths(vcsPathsForProvenances[scanResult.provenance]!!)
+            )
+        }
```

Comment:
I propose to combine the two like

        }.mapNotNullTo(mutableSetOf()) { scanResult ->
            vcsPathsForProvenances[scanResult.provenance]?.let {
                scanResult.copy(summary = scanResult.summary.filterByPaths(it))
            }
        }

which also avoids the `!!`.